dir.targets

```diff
@@ -28,6 +28,15 @@
     <TargetFrameworkProfile></TargetFrameworkProfile>
     <!-- We set this property to avoid MSBuild errors regarding not setting TargetFrameworkProfile (see above line) -->
     <PortableNuGetMode>true</PortableNuGetMode>
+
+    <!--Start with NonUnixTests if we aren't on Windows and overwrite with a specific OS, if we know what it is-->
+    <TestFilterCategory Condition="'$(OS)' != 'Windows_NT'">-notrait category=nonunixtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'OSX'">-notrait category=nonosxtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'Windows_NT'">-notrait category=nonwindowstests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'Linux'">-notrait category=nonlinuxtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'FreeBSD'">-notrait category=nonfreebsdtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'NetBSD'">-notrait category=nonnetbsdtests</TestFilterCategory>
+    <TestCommandLine>$(TestCommandLine) $(TestFilterCategory)</TestCommandLine>
```

Comment:
I'm grabbing a copy of this code change to dir.targets, as-is, for a change I am currently working on.  I'll be sure to cc you on the PR so that you are aware of any conflicts if you make additional changes.  This change is simpler than the build tools change I was considering, only because coordination between buildtools and corefx won't be required.  Is there a plan to move this into BuildTools?  This seems somewhat redundant and likely to cause conflicts at some point given that these are additive and keying off of different properties...

https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L60